### ISSUE TITLE
Better RegExp rule about letter case

### ISSUE THREAD

**Issue by Jack-Works** [sindresorhus/eslint-plugin-unicorn#1476]
With better regexp rule:

![image](https://user-images.githubusercontent.com/5390719/128815952-47ba8bfb-b5bf-4d45-8df0-b1128504a197.png)

But as you can see, `ThisIsATest` is more readable than `thisisatest`. I suggest to add a new rule to do not warn about the letter cases.

**Comment by fisker** [comment]
I think `eslint-disable` make sense for this case.

**Comment by zloirock** [comment]
Because of non-configurability of this rule, for my cases, I replaced it with `eslint-plugin-regexp`.

**Comment by Jack-Works** [comment]
Why it's non-configurable? 👀 Can't it ignore the suggestion that only differ in case?

**Comment by zloirock** [comment]
It just compares regex with the result of calling [`clean-regexp`](https://github.com/samverschueren/clean-regexp) on it. Sure, for your case can be added a flag for apply `toLowerCase` on source with `i` flag, but it's too specific and more complex options to this rule can't be added.

**Comment by zloirock** [comment]
Hmm, seems `regexp-tree` `.optimize` method contains [some options](https://github.com/DmitrySoshnikov/regexp-tree/tree/master/src/optimizer#transforms) that can be blacklisted, including `charCaseInsensitiveLowerCaseTransform`, maybe makes sense to expose this possibility?

**Comment by Jack-Works** [comment]
> Hmm, seems `regexp-tree` `.optimize` method contains [some options](https://github.com/DmitrySoshnikov/regexp-tree/tree/master/src/optimizer#transforms) that can be blacklisted, including `charCaseInsensitiveLowerCaseTransform`, maybe makes sense to expose this possibility?

Yes, I believe so, as my top threads said, sometimes uppercase can help to improve readability.

**Comment by fisker** [comment]
I'm fine to add this option.

**Comment by sindresorhus** [comment]
The better regex rule was removed, so closing.
